### PR TITLE
Switched a few ports to OneTBB instead TBB

### PR DESCRIPTION
**Pull request by catap**
#### Description

OneTBB is a new and supported version for Threading Building Blocks API.

Here I've switched a few ports from `tbb` to `onetbb`, some of them were also updated.

I've stop here with hope that GitHub CI timeout is enough to build all touched ports.

###### Type(s)

- [x] bugfix
- [x] enhancement

###### Tested on

macOS 12.6 21G115 x86_64
Xcode 14.1 14B47b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: graphics/vtk-m/Portfile [hunk 1]
```diff
@@ -59,11 +59,11 @@ variant openmp description {Enable OpenMP support} {
     configure.ldflags-append -L${prefix}/lib/libomp -lomp
 }
 
-variant tbb description {Enable Intel Threading Building Blocks support} {
+variant tbb conflicts onetbb description {Enable oneAPI Threading Building Blocks support} {
```
Comment (reneeotten) [inline graphics/vtk-m/Portfile:62]: why do you add a conflict here, it seems you didn't add a "onetbb" variant? Last time I checked the software as far as I could tell it wasn't compatible with oneTBB - what did you do to verify that it is?

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: graphics/vtk-m/Portfile (hunk 1 above)
Comment (catap) [reply graphics/vtk-m/Portfile:62]: @reneeotten thanks to catching this up. Original version was to add to add `onetbb` and after that I've simplified it. But seems to forgot to clean it up.

Comment (reneeotten) [reply graphics/vtk-m/Portfile:62]: well that was the minor point here 😉 - how did you confirm that this change is functional/correct?

Comment (catap) [reply graphics/vtk-m/Portfile:62]: How I've tested it. I've installed it as `vtk-m +tbb` to confirm that it built and links, it has built and it had linked sucefully.

Do you think it requires more testing?

Comment (catap) [reply graphics/vtk-m/Portfile:62]: (To save CI time, I'll made push to fix this typo as soon as CI had happened. Maybe I will include some future fixes)

**Comment by catap** [comment]
@reneeotten for example Debian ships OneTBB as `libtbb`: https://packages.debian.org/sid/libtbb-dev

**Comment by catap** [comment]
@ra1nb0w I've built `kicad @6.0.9_1` and suceffuly opened demo project `microwave`.

**Comment by catap** [comment]
6.5 hours isn't enough. CI had failed by timout.

I've made a force push to fix @reneeotten remarks.

I can confirm that I've tried to build all this ports on my laptop, and this PR is read for review and to be merged.

**Comment by catap** [comment]
@reneeotten I've excluded `vtk-m` and `oce`; maybe now it passes CI.

**Comment by catap** [comment]
> For now it works perfectly fine and I don't see a reason to switch to `oneTBB`

Right now it is impossible to instal `onetbb` and `tbb` because the first one is a new version of the last one :) And `https://github.com/intel/tbb` redirects to `https://github.com/oneapi-src/oneTBB`.

This switch more than upgrade of version, but if you would like to keep your port not updated, fine with me.

**Comment by barracuda156** [comment]
@catap I get a failure with `oneTBB` as `+universal` on 10.6.8 Intel: https://trac.macports.org/ticket/66379

**Comment by reneeotten** [comment]
> Right now it is impossible to instal `onetbb` and `tbb` because the first one is a new version of the last one :) And `https://github.com/intel/tbb` redirects to `https://github.com/oneapi-src/oneTBB`.

then an interested party should probably invest the time on solving *that* problem. The title of the PR is "switched a few ports...." - that is not going to help then, it would need to be done for all ports otherwise the problem that currently the `tbb` and `onetbb` ports cannot be installed simultaneously remains an issue.

**Comment by barracuda156** [comment]
> then an interested party should probably invest the time on solving *that* problem. The title of the PR is "switched a few ports...." - that is not going to help then, it would need to be done for all ports otherwise the problem that currently the `tbb` and `onetbb` ports cannot be installed simultaneously remains an issue.

@reneeotten In a comparative perspective, it is more desirable to have a port depend on `oneTBB` rather than `TBB`, since the former is the only one supported by the upstream (and assuming my PR for PPC support is merged).
Of course, having both is still better, if feasible.

**Comment by cjones051073** [comment]
More needs to be done than is currently here, I am afraid.

Either you need to convert *all* ports to use `oneTBB` instead of `tbb`, and then declare `tbb` as `replaced_by` `oneTBB`, or if we need to have a mixture, then the ports need updating to make the co-exist together, so install to independent locations. i.e. have `oneTBB` instead to the main prefix, but for the ports that still need it have `tbb` available at an alternative location.

**Comment by barracuda156** [comment]
> More needs to be done than is currently here, I am afraid.
> 
> Either you need to convert _all_ ports to use `oneTBB` instead of `tbb`, and then declare `tbb` as `replaced_by` `oneTBB`, or if we need to have a mixture, then the ports need updating to make the co-exist together, so install to independent locations. i.e. have `oneTBB` instead to the main prefix, but for the ports that still need it have `tbb` available at an alternative location.

@cjones051073 Just for a note, this PR does not _introduce_ the problem: we already have some ports depending on `oneTBB`: https://ports.macports.org/port/onetbb/details

**Comment by catap** [comment]
I've marked it to draft because I need some time to migrate all ports to OneTBB.

Stay tuned!

**Comment by catap** [comment]
Cleaner approach: https://github.com/macports/macports-ports/pull/16915

**Pull request closed without merging**